addons/event_sale/models/sale_order.py

```diff
@@ -22,8 +22,15 @@ def write(self, vals):
     def action_confirm(self):
         res = super(SaleOrder, self).action_confirm()
         for so in self:
+            so_lines_missing_events = self.env['sale.order.line']
             if not any(line.product_type == 'event' for line in so.order_line):
                 continue
+            for line in so.order_line:
+                if (line.product_type == 'event' or line.product_type == 'event_booth') and not line.event_id:
```

Comment:
"event_booth" product type is introduced in the "event_booth_sale" module, should not be tested here.

Reply:
when both "event_sale" and "event_booth_sale" modules are installed, we first execute functions in "event_sale" and then to "event_booth_sale". This means that if we want to show a error specifying all the unconfigured events and event_booths, we will have to do it in "event_sale" module. Otherwise our error will show only unconfigured eventsand not unconfigured event booths.

Reply:
That's not very consistent then because in event_booth_sale you check for missing "event_booth_pending_ids" and here you check for missing "event_id" for events AND booth.

In any case it's not very modular to check a value that cannot exist without another module installed (even if it would not crash).

I suggest that both modules do their own checks, and if the user has both events and booths misconfigured in his SO, he will click confirm once, get the error about events not being configured, configure them, then click confirm again, get the error about booths, configure them, then he's good to go.

It would be extremely rare to have both anyway, and in future cases, if we add this logic for other things (misconfigured products for example), we will not be able to do all checks about all configurations in all modules anyway.